services/supervisor/tests/supervisor/supervisor_config.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tests for the `Supervisor` service configuration, including
+//! `Supervisor` service initialization, using `Configure` interface
+//! and API endpoints associated with configuration.
+
+use exonum::{
+    blockchain::config::InstanceInitParams,
+    runtime::{rust::ServiceFactory, InstanceSpec, SnapshotExt, SUPERVISOR_INSTANCE_ID},
+};
+use exonum_merkledb::BinaryValue;
+use exonum_testkit::{ApiKind, TestKit, TestKitBuilder};
+
+use exonum_supervisor::{
+    mode, supervisor_name, ConfigChange, ConfigPropose, Schema, ServiceConfig, Supervisor,
+    SupervisorConfig,
+};
+
+use crate::{config_api::create_proposal, utils::CFG_CHANGE_HEIGHT};
+
+/// Asserts that current supervisor configuration equals to the provided one.
+fn assert_supervisor_config(testkit: &TestKit, config: SupervisorConfig) {
+    let snapshot = testkit.snapshot();
+    let current_config = Schema::new(snapshot.for_service(supervisor_name()).unwrap())
+        .configuration
+        .get()
+        .unwrap();
+    assert_eq!(current_config, config);
+}
+
+/// Checks that initial configuration providers (`Supervisor::simple_config()` and
+/// `Supervisor::decentralized_config()`) provide correct values and Supervisor
+/// loads them as expected.
+#[test]
+fn initial_configuration() {
+    // Check for simple mode.
+    let testkit = TestKitBuilder::validator()
+        .with_rust_service(Supervisor)
+        .with_artifact(Supervisor.artifact_id())
+        .with_instance(Supervisor::builtin_instance(Supervisor::simple_config()))
+        .create();
+    assert_supervisor_config(
+        &testkit,
+        SupervisorConfig {
+            mode: mode::Mode::Simple,
```

Comment:
Nit: Is module prefixing really warranted here? Looks weird, IMO. Also, shouldn't it be possible to use `Supervisor::*_config()` methods throughout the tests?